docs/DeveloperGuide.md

```diff
@@ -6,8 +6,248 @@
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+## Design
 
+### Add Housekeeper Command
+
+The objective of the `AddHousekeeperCommand` is to take in user input and spilt it into two parts which is the name and age of a Housekeeper. These details make up parts of the Housekeeper profile. With the name and age derived, this information will be added into a new Housekeeper object, which will be recorded into the list of housekeeper. The class diagram below depicts how the `AddHousekeeperCommand` interacts with other classes.
+
+![class](team/falicia_addHousekeeperCommand/classAddHousekeeper.jpg)
+
+Below is an example showing the usage of `AddHousekeeperCommand`.
+1. Command given from the user: `add housekeeper susan / 33`. This command meant to add a new housekeeper called susan who is 33 years old.
+2. The parse method from `CommandParser` will run parse to create `AddHousekeeperCommand` and would be return to Duke.
+3. The execute method in `AddHousekeeperCommand` will be performed. It first checks with the housekeeperList if the name of the housekeeper to be added has been recorded into the list. If it has not been recorded, housekeeperList would then add this new user into the records. Ui will be called to print a confirmation to show that the housekeeper has been entered into the list.
+5. If housekeeper exist, the housekeeper profile would not be added into the list.
+
+### Add Item Command
+The objective of the AddItemCommand is to allow the user to add a new item to the list of items found within the inventory. It takes in the user input and spilt it up into two parts which are the name of the item to be added and its pax. These two information would then be used to create an Item Object and the Item Object would be saved into a list of items.
+
+Below is the partial class diagram detailing the design of the Add Item Command Class as well as its interactions with the various other classes required to execute the Add Item Command.
+![alt text](team/SiewYangZhi_addItemCommand/AddItemCommandClassDiagram.png)
```

Comment:
For the diagram of `Add Item Command`, I think one method named `public UI` in `UI` might need to delete `public` as it's already shown as `+` in the class diagram.